### PR TITLE
Fix bugs, add features to work with SoftHSM v2.

### PR DESCRIPTION
**Pull request by epim**
A lot of changes this go-around. I suspect the unit tests worked
previously due to bugs in SoftHSM v1. The biggest problems were
TestFindObject and TestGetAttributeValue failing because they
depended on nonexistent objects. SoftHSM v1 must not have
enforced session based tokens correctly.

* Add token_persistent to generateRSAKeyPair to set the value of
  CKA_TOKEN, which determines whether or not a token will be
  persistent (stored in the DB) or session based. This is to allow
  for persistent token unit tests to be added down the line.

* Add token_label to generateRSAKeyPair, to set the label string
  for the token. Useful for debugging which functions are leaving
  tokens lying around.

* Add generateRSAKeyPair to TestGetAttributeValue and
  TestFindObject. These tests were failing because there simply
  weren't any tokens around for them to find. Other tests didn't
  leave anything lying around, because they only created session
  based tokens.

* split testDestroyObject into TestDestroyObject and
  destroyObject. This allows for destroyObject to be re-used by
  other functions that want to play with persistent tokens, as
  well as enabling tests for creating and destroying persistent
  tokens.

* Enable TestDestroyObject to create and destroy persistent tokens in
  the HSM

* Add 'CKF_RW_SESSION' to getSession, to allow for persistent
  token management.

* remove erroneous use of SO credentials in ExampleSign. By design the account model in PKCS11 practices the Principal of Least Privilege, and neither the Admin or Security Officer may provision or read the contents of tokens.

### CODE REVIEW

File: pkcs11_test.go [hunk 1]
```diff
@@ -198,20 +204,45 @@ func TestDigestUpdate(t *testing.T) {
...
+				destroyed upon termination of the session.
+	outputs: creates persistent or ephemeral tokens within the HSM.
+	returns: object handles for public and private keys.
+	*/
+			
 	publicKeyTemplate := []*Attribute{
 		NewAttribute(CKA_CLASS, CKO_PUBLIC_KEY),
 		NewAttribute(CKA_KEY_TYPE, CKK_RSA),
-		NewAttribute(CKA_TOKEN, false),
+		NewAttribute(CKA_TOKEN, token_persistent),
```
Comment (miekg) [inline pkcs11_test.go]: Style nit: names like this should be camelCase: tokenPersistent

**Comment by miekg** [comment]
Thanks! Looks good. 
Only the style nit wrt to naming vars.

**Comment by epim** [comment]
Thanks for looking! I'll camel case those vars.

**Comment by epim** [comment]
PR Updated with camelcased vars

**Pull request merged**